### PR TITLE
implement prefixer POC

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This is 2.0 material

`setPragma` --> `setup`

This new setup function accepts a second argument named `prefixer` which can be passed the prefixer exported by `goober-autoprefixer`.

This method given a key and value will prefix when needed thanks to the awesome [`tiny-css-prefixer`](https://github.com/kitten/tiny-css-prefixer)

### CODE REVIEW

File: packages/goober-autoprefixer/__tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+describe('goober-autoprefixer', () => {
+    it('can prefix values', () => {
+        const output = parse({ position: 'sticky' }, 'div');
+        expect(sanitize(output)).toEqual('div{position: -webkit-sticky, sticky;}');
+    });
+
+    it('can prefix properties', () => {
+        const output = parse({ appearance: 'none' }, 'div');
+        expect(sanitize(output)).toEqual(`div{appearance: none;-moz-appearance: none;
```
Comment (JoviDeCroock) [inline packages/goober-autoprefixer/__tests__/index.test.js]: TODO: improve this

**Comment by cristianbote** [comment]
@JoviDeCroock this is sooooo awesome! 🎉 thank you! ;)

File: packages/goober-autoprefixer/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "goober-autoprefixer",
+  "version": "1.0.0",
+  "main": "./dist/hooked-form.js",
+  "browser": "./dist/hooked-form.umd.js",
+  "umd:main": "./dist/hooked-form.umd.js",
+  "jsnext:main": "./dist/hooked-form.modern.js",
+  "module": "./dist/hooked-form.module.js",
```
Comment (cristianbote) [inline packages/goober-autoprefixer/package.json]:
```suggestion
  "module": "./dist/goober-autoprefixer.module.js",
```
😃

**Comment by cristianbote** [comment]
Browsing through the changes, I can't but think about how to ship this? 🤔 Should it be a separate package? like:
```
import { h } from 'preact';
import { styled, setPragma } from 'goober';
import  'goober-autoprefixer';

setPragma(h);
```

Or since we're gonna have to expand the `setPragma` into a `setup` function that would accept more  options maybe prefixer could be of those? 
```
import { h } from 'preact';
import { styled, setup } from 'goober';
import { prefixer }  'goober-autoprefixer';

setup({ h, prefixer });
```
🤔

**Comment by JoviDeCroock** [comment]
Almost there, only the `cjs` bundle is a byte too big

**Comment by cristianbote** [comment]
Awesome!💪 Watching closely! 😎 I managed to shave 2B as well from master. Will send a PR later to see what fails 😬

**Comment by cristianbote** [comment]
@JoviDeCroock this is so freaking awesome! 🎉 thank you!

**Pull request merged**